### ISSUE TITLE
Lagging when moving items between inventory and ExtraSlots

### ISSUE THREAD

**Issue by Jeesson** [shudnal/ExtraSlots#3]
I'm experiencing significant lag spikes when transferring items between the inventory and ExtraSlots as well as when opening inventory. Every time I move an item in or out, there’s a noticeable freeze that disrupts the gameplay experience. This issue may be related to performance optimization or item handling in the ExtraSlots feature.

After some troubleshooting, I believe the issue might be related to MyLittleUI. Once I removed this mod, the lag spikes disappeared, and the item transfer became smooth again.

Below is the list of mods currently installed, which might help in identifying any compatibility issues.
```
EpicLoot v0.10.6
PassivePowers v1.1.2
SteadyRegeneration v1.0.3
TimedItemDestruction v1.0.3
Sailing v1.1.7
Smoke_Collision v1.8.0
CraftyCartsRemake v3.1.6
CoolerWorkbench v2.0.0
Backpacks v1.3.5
Jotunn v2.21.3
XPortal v1.2.21
InstantMonsterLootDrop v0.5.0
Recycle_N_Reclaim v1.3.5
Exploration v1.0.3
UsefulTrophies v2.2.2
HipWeapons v1.1.0
SimpleElevators v1.2.7
BetterLanterns v1.1.0
BedRules v2.0.2
DynamicStoragePiles v0.6.1
StartupAccelerator v1.0.3
Compass v1.0.1
AzuCraftyBoxes v1.5.6
SeedBed v1.2.4
CreatureLevelAndLootControl v4.6.2
AdventureBackpacks v1.7.9
ExtraSlots v1.0.2
MyLittleUI v1.1.16
```

**Comment by shudnal** [comment]
I'll check MyLittleUI for potential issue. 

Are you sure you don't have a nocost enabled and your item list is not filled with all items? It can cause performance issue with crafting list.

**Comment by Jeesson** [comment]
Yes, I'm sure. I'm running a standard survival world, and we have already defeated five bosses.

**Comment by Jeesson** [comment]
Lag occurs when receiving an item with the inventory open. The issue might be related to crafting sorting, as similar severe lag spikes also appear when opening workbenches, even with an empty crafting field.

**Comment by shudnal** [comment]
It should be fixed in 1.1.18. I've made major improvement on multicraft and I have no idea how to optimize performance further.

You could still feel small lag if you use any "craft from containers" mod and you're surrounded by huge amount of chests but that's because to calculate max amount of multicraft it has to check all containers in the area. 

If that said mod doesn't handle that behaviour well (which is easy doable btw) then the lag will still be there. But if mod handles it well it should work as fast as regular player inventory.
